tiatoolbox/wsicore/wsireader.py

```diff
@@ -68,6 +69,60 @@ def is_tiled_tiff(path: pathlib.Path) -> bool:
     return tif.pages[0].is_tiled
 
 
+def is_zarr(path: pathlib.Path) -> bool:
+    """Check if the input is a Zarr file.
+
+    Args:
+        path (pathlib.Path):
+            Path to the file to check.
+
+    Returns:
+        bool:
+            True if the file is a Zarr file.
+
+    """
+    path = pathlib.Path(path)
+    try:
+        _ = zarr.open(path, mode="r")
+        return True
+    # skipcq: PYL-W0703
+    except Exception:  # noqa: PIE786
+        return False
+
+
+def is_ngff(path: pathlib.Path) -> bool:
```

Comment:
This is not used anywhere. @John-P were you planning on using this when deciding whether to use the NGFFWSIReader or not?

Reply:
Yes, that plan is to use this to validate if it is an NGFF formatted Zarr or not.

Reply:
I may also add support for loading a generic Zarr array also (non NGFF formatted).

Reply:
This is now being used to check the Barr in `WSIReader.open`